pkg/controller/datavolume/controller-base.go

```diff
@@ -481,6 +481,10 @@ func (r *ReconcilerBase) syncDvPvcState(log logr.Logger, req reconcile.Request,
 		r.handlePrePopulation(syncState.dvMutated, syncState.pvc)
 	}
 
+	if err = updateDataVolumeDefaultInstancetypeLabels(r.client, &syncState); err != nil {
```

Comment:
@arnongilboa - if you have a second, does updating the `DataVolume` at this point make sense or should this be done elsewhere? I'm also ignoring errors here given you previously highlighted that the `PVC` and/or `DataSource` might not exist yet.

Reply:
Putting it right after the other DV updates (before `if syncState.pvc != nil`) is better.
btw why return error if you ignore it without even checking which error is it?

Reply:
Only to log the error, happy to move that down into `updateDataVolumeDefaultInstancetypeLabels` if you'd prefer.

Reply:
I think that if !IsNotFound(err) you should `return syncState, err`